### ISSUE TITLE
pod lib lint - ERROR | [watchOS] unknown: Encountered an unknown error (CocoaPods could not find compatible versions for pod "FirebaseCore"

### ISSUE THREAD

**Issue by Bino90** [CocoaPods/CocoaPods#9524]
# Report

## What did you do?

pod lib lint MYPOD.podspec --allow-warnings

## What did you expect to happen?

no errors

## What happened instead?

- ERROR | [watchOS] unknown: Encountered an unknown error (CocoaPods could not find compatible versions for pod "FirebaseCore":

In my .podspec i set that firebase should only be available on all other platforms .....

Can anyone help?

## CocoaPods Environment

```
   CocoaPods : 1.8.4
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin19]
    RubyGems : 3.0.3
        Host : Mac OS X 10.15.3 (19D76)
       Xcode : 11.3.1 (11C504)
         Git : git version 2.21.1 (Apple Git-122.3)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : master - git - https://github.com/CocoaPods/Specs.git @ a9f329b380ba8b6cc66a7d9542d0d82d0c913b40
               trunk - CDN - https://cdn.cocoapods.org/
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.4.1
cocoapods-try         : 1.1.0
```

**Comment by paulb777** [comment]
The error message points in the right direction. FirebaseCore requires at least watchOS 6.0 and the podspec is at 3.0.  See https://github.com/CocoaPods/Specs/blob/master/Specs/8/b/d/FirebaseCore/6.6.1/FirebaseCore.podspec.json#L21

**Comment by Bino90** [comment]
yeah right but the firebase dependency is only for iOS, tvOS and OSX(macOS) not for watchOS
    sp.ios.dependency 'Firebase/Crashlytics'
    sp.tvos.dependency 'Firebase/Crashlytics'
    sp.osx.dependency 'Firebase/Crashlytics'

**Comment by dnkoutso** [comment]
@paulb777 any further updates here?

**Comment by paulb777** [comment]
@dnkoutso Looks like a CocoaPods bug since it shouldn't be checking the `watchos` version when `watchos` is not included in the platform list.

**Comment by dnkoutso** [comment]
ah I see...interesting will keep it open then.

**Comment by paulb777** [comment]
@Bino90 I took an initial stab at this and didn't repro. Would you send a repro example?

Also, a workaround may be running `pod lib lint` separately for the different platforms with the `--platforms` options.

**Comment by dnkoutso** [comment]
I was able to repro this a while ago but I forgot again how i did it...still keeping this for 1.10 for a bit but I think we will punt if the time comes to release 1.10.
